Crash while proving termination for function with inhale-exhale expression in postcondition

The following program crashes Silicon:
```
function foo(): Bool

function bar(): Int
  decreases
  ensures [foo(), true]
```

<details><summary>Stack trace</summary>
<p>

```
Caused by: java.util.NoSuchElementException: key not found: $condInEx
        at scala.collection.MapOps.default(Map.scala:274)
        at scala.collection.MapOps.default$(Map.scala:273)
        at scala.collection.AbstractMap.default(Map.scala:405)
        at scala.collection.MapOps.apply(Map.scala:176)
        at scala.collection.MapOps.apply$(Map.scala:175)
        at scala.collection.AbstractMap.apply(Map.scala:405)
        at viper.silicon.state.MapBackedStore.apply(Store.scala:46)
        at viper.silicon.rules.evaluator$.eval2(Evaluator.scala:173)
        at viper.silicon.rules.evaluator$.eval3(Evaluator.scala:134)
        at viper.silicon.rules.evaluator$.eval(Evaluator.scala:95)
        at viper.silicon.rules.executor$.follow(Executor.scala:69)
        at viper.silicon.rules.executor$.$anonfun$follows$14(Executor.scala:195)
        at viper.silicon.interfaces.VerificationResult.combine(Verification.scala:45)
        at viper.silicon.rules.executor$.$anonfun$follows$13(Executor.scala:195)
        at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:183)
        at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:179)
        at scala.collection.immutable.List.foldLeft(List.scala:79)
        at viper.silicon.rules.executor$.follows(Executor.scala:189)
        at viper.silicon.rules.executor$.$anonfun$exec$1(Executor.scala:217)
        at viper.silicon.rules.executor$.execs(Executor.scala:319)
        at viper.silicon.rules.executor$.$anonfun$execs$1(Executor.scala:317)
        at viper.silicon.rules.executor$.$anonfun$exec$25(Executor.scala:327)
        at viper.silicon.rules.executor$.$anonfun$exec2$1(Executor.scala:336)
        at viper.silicon.rules.executor$.exec2(Executor.scala:361)
        at viper.silicon.rules.executor$.exec(Executor.scala:325)
        at viper.silicon.rules.executor$.execs(Executor.scala:316)
        at viper.silicon.rules.executor$.exec(Executor.scala:216)
        at viper.silicon.rules.executor$.exec(Executor.scala:207)
        at viper.silicon.supporters.DefaultMethodVerificationUnitProvider$methodSupporter$.$anonfun$verify$16(MethodSupporter.scala:113)
        at viper.silicon.rules.executionFlowController$.$anonfun$locally$1(ExecutionFlowController.scala:99)
        at viper.silicon.rules.executionFlowController$.locallyWithResult(ExecutionFlowController.scala:62)
        at viper.silicon.rules.executionFlowController$.locally(ExecutionFlowController.scala:99)
        at viper.silicon.supporters.DefaultMethodVerificationUnitProvider$methodSupporter$.$anonfun$verify$15(MethodSupporter.scala:112)
        at viper.silicon.interfaces.NonFatalResult.$amp$amp(Verification.scala:84)
        at viper.silicon.supporters.DefaultMethodVerificationUnitProvider$methodSupporter$.$anonfun$verify$11(MethodSupporter.scala:112)
        at viper.silicon.rules.producer$.produceTlcs(Producer.scala:143)
        at viper.silicon.rules.producer$.produces(Producer.scala:131)
        at viper.silicon.supporters.DefaultMethodVerificationUnitProvider$methodSupporter$.$anonfun$verify$9(MethodSupporter.scala:101)
        at viper.silicon.rules.executionFlowController$.$anonfun$locally$1(ExecutionFlowController.scala:99)
        at viper.silicon.rules.executionFlowController$.locallyWithResult(ExecutionFlowController.scala:62)
        at viper.silicon.rules.executionFlowController$.locally(ExecutionFlowController.scala:99)
        at viper.silicon.supporters.DefaultMethodVerificationUnitProvider$methodSupporter$.verify(MethodSupporter.scala:100)
        at viper.silicon.verifier.DefaultMainVerifier.$anonfun$verify$8(DefaultMainVerifier.scala:265)
        at viper.silicon.verifier.VerificationPoolManager$WorkerAwareForkJoinTask.compute(VerificationPoolManager.scala:153)
        at viper.silicon.verifier.VerificationPoolManager$WorkerAwareForkJoinTask.compute(VerificationPoolManager.scala:146)
        at java.base/java.util.concurrent.RecursiveTask.exec(RecursiveTask.java:100)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

</p>
</details> 

I believe this happens while proving termination for the bar function since in the debugger I can see that it's trying to verify the following method:

```
// decreases 
method bar_posts_termination_proof()
{
  var $result: Int
  if ($condInEx) {
    assert false
  }
}
```

--- a/src/main/scala/viper/silver/plugin/standard/termination/transformation/FunctionCheck.scala
+++ b/src/main/scala/viper/silver/plugin/standard/termination/transformation/FunctionCheck.scala
@@ -98,7 +98,7 @@ trait FunctionCheck extends ProgramManager with DecreasesCheck with ExpTransform
 
         if (proofMethodBody != EmptyStmt) {
           val proofMethod = Method(proofMethodName, f.formalArgs, Nil, f.pres, Nil,
-            Option(Seqn(Seq(proofMethodBody), Seq(resultVariable))()))(info = f.info)
+            Option(Seqn(Seq(proofMethodBody), Seq(resultVariable, context.conditionInEx.get))()))(info = f.info)
 
           Seq(proofMethod)
         } else {

--- a/src/test/resources/all/issues/silicon/0883.vpr
+++ b/src/test/resources/all/issues/silicon/0883.vpr
@@ -1,3 +1,7 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+
 field f: Int
 
 @moreJoins("2")

--- a/src/test/resources/all/issues/silicon/0886.vpr
+++ b/src/test/resources/all/issues/silicon/0886.vpr
@@ -1,0 +1,10 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+
+function foo(): Bool
+  decreases
+
+function bar(): Int
+  decreases
+  ensures [foo(), true]

